sarathi/model_executor/attention/base_attention_wrapper.py

```diff
@@ -1,31 +1,32 @@
 from abc import ABC, abstractmethod
-from typing import List, Optional, Tuple, Union
+from typing import Dict, List, Optional, Tuple, Union
 
 import torch
 
-from sarathi.config import ModelConfig, ParallelConfig
+from sarathi.config import CacheConfig, ModelConfig, ParallelConfig
 from sarathi.core.datatypes.sequence import SequenceMetadata
 from sarathi.metrics.constants import OperationMetrics
 from sarathi.metrics.cuda_timer import CudaTimer
 
-
 class BaseAttentionWrapper(ABC):
-    _inst = None
 
-    def init(
+    def __init__(
         self,
         model_config: ModelConfig,
         parallel_config: ParallelConfig,
-        block_size: int,
+        cache_config: CacheConfig,
         device: torch.device,
     ):
         self.device = device
         self.num_q_heads = model_config.get_num_q_heads(parallel_config)
         self.num_kv_heads = model_config.get_num_kv_heads(parallel_config)
         self.head_dim = model_config.get_head_size()
         self.dtype = model_config.dtype
-        self.block_size = block_size
-        self._timers = {}
+        self.block_size = cache_config.block_size
+        self.num_layers = model_config.get_num_layers(parallel_config)
+        self.num_gpu_blocks: int = cache_config.num_gpu_blocks
+        self.gpu_cache: Optional[List[torch.Tensor]] = None
+        self._timers:  Optional[Dict[Tuple[OperationMetrics, int], CudaTimer]] = {}
```

Comment:
nit: i guess now it makes sense to just call this self.timers

Reply:
Why is that? As I understand it, leading _ conventionally refers to private members. It is still a private member right?

Reply:
all other properties are here are also private. earlier we were using convention where all private properties are marked with _, however, it conflicts with vllms style creating a lot of conflicts. so now, we are moving to a pattern where private methods are marked with _ but not private properties

Reply:
Understood, thanks! Made the change.